File: cypress/support/pages/cartPage.js
class CartPage {
  elements = {
    /** Recebe o nome do produto clicado e o encontra entre os produtos listados no carrinho
     * @param {string} productTitle - Nome do produto a ser verificado no carrinho
     * @returns {Cypress.Chainable<JQuery<HTMLElement>>} - Retorna o elemento encontrado
     */
    productTitleElement: (productTitle) => cy.contains("h2", productTitle),

    /** Botão 'Buy Products' para continuar com o checkout */
    buyProductsButton: () => cy.contains(".btn", "Buy Products"),
  };

  //Ações

  /** Verifica se o produto está listado no carrinho
   * @param {string} productTitle - Nome do produto a ser verificado no carrinho
   * @returns {CartPage} - Retorna a instância da classe CartPage para encadeamento de métodos
   */
  verifyProductInCart(productTitle) {
    this.elements.productTitleElement(productTitle).should("be.visible");
    return this;
  }

  /** Clica no botão 'Buy Products' para continuar com o checkout
   * @returns {CartPage} - Retorna a instância da classe CartPage para encadeamento de métodos
   */
  clickBuyProductsButton() {
    this.elements.buyProductsButton().click();
    return this;
  }
}

export default new CartPage();

File: pages/cart_page.py
from playwright.sync_api import Page, expect


class CartPage:
    def __init__(self, page: Page):
        self.page = page
        self.buy_products_button = page.get_by_role("button", name="Buy Products")

    # --- Elementos (parametrizados) ---

    def product_title_element(self, product_title: str):
        return self.page.get_by_role("heading", name=product_title, level=2)

    # --- Ações ---

    def verify_product_in_cart(self, product_title: str):
        expect(self.product_title_element(product_title)).to_be_visible()
        return self

    def click_buy_products_button(self):
        self.buy_products_button.click()
        return self
